src/sentry/static/sentry/app/components/pill.jsx

```diff
@@ -29,19 +29,80 @@ class Pill extends React.Component {
   };
 
   render() {
-    const {name, children, className, ...props} = this.props;
-    const [extraClass, renderedValue] = this.renderValue();
+    const {name, children, ...props} = this.props;
+    const [renderedValue] = this.renderValue();
 
     return (
-      <li className={classNames(className, extraClass)} {...props}>
-        <span className="key">{name}</span>
-        <span className="value">
+      <StyledPill {...props}>
+        <PillName>{name}</PillName>
+        <PillValue>
           {renderedValue}
           {children}
-        </span>
-      </li>
+        </PillValue>
+      </StyledPill>
     );
   }
 }
 
+const StyledPill = styled('li')`
+  white-space: nowrap;
+  margin: 0 10px 10px 0;
+  display: flex;
+  border: 1px solid #d0c9d7;
+  border-radius: 3px;
+  box-shadow: 0 1px 2px rgba(0, 0, 0, 0.04);
+  line-height: 1.2;
+  max-width: 100%;
+  &.true .value {
```

Comment:
So instead of nesting these selectors... `.value` refers to what is now `<PillValue>` and `true/false` comes from `extraClass`

What we should do instead is pass extraClass as a prop to `<PillValue>` and handle the CSS in our `PillValue` component.